### PR TITLE
fix: corrections and improvements to IRQ how-to

### PR DESCRIPTION
**Pull request by farshidtz**
Fixing a few issues and improving the writing.

The how-to can actually benefit from a major refactoring to use `smp_affinity_list` instead of `smp_affinity`, providing more understandable examples.

### CODE REVIEW

**Review by locnnil — COMMENTED** [review]
Changes looks good. I found it also important for this session to explain that [SMI](https://wiki.linuxfoundation.org/realtime/documentation/howto/debugging/smi-latency/smi) cannot be reassigned and return an `Input/output error` if tried to be reassign.
As said in the LF documentation: 

> The operating system has absolutely no control over when SMIs will happen or how long the CPU will subsequently stay in system management mode. 

Maybe add a Note session for this.

**Comment by farshidtz** [comment]
> Changes looks good. I found it also important for this to explain that [SMI](https://wiki.linuxfoundation.org/realtime/documentation/howto/debugging/smi-latency/smi) cannot be reassigned and return an `Input/output error` if attempted to reassign. As said in the LF documentation:
> 
> > The operating system has absolutely no control over when SMIs will happen or how long the CPU will subsequently stay in system management mode.
> 
> Maybe add a Note session for this.

This would be a good addition.
Please consider submitting this via a separate PR, including first an explanation of what SMIs are.

**Review by locnnil — APPROVED** [review]

File: docs/how-to/tune-irq-affinity.rst [hunk 1]
```diff
@@ -19,35 +19,35 @@ The most common `kernel parameters`_ to handle IRQ interrupts are:
 - `isolcpus`_ used to specify CPUs to be isolated from the general `SMP`_ balancing and scheduler algorithms. 
 
 
-The `IRQ affinity`_ on  other hand is a way to tune the system interruptions
+The `IRQ affinity`_ on other hand is a way to tune the system interruptions
```
Comment (AnneCYH) [inline docs/how-to/tune-irq-affinity.rst:22]:
```suggestion
The `IRQ affinity`_ on the other hand is a way to tune the system interruptions
```

File: docs/how-to/tune-irq-affinity.rst [hunk 2]
```diff
@@ -60,22 +60,22 @@ In this case, the system has 20 CPUs available.
 The best way to tune a system is to isolate one or more CPUs to be used to run 
 the real-time application and the others to handle the IRQs and kthreads.
 
-First check which IRQ is being handled by the CPU, or sets of CPUs that you want
-to isolate. This can do either manually checking:
+To check which CPUs are set for handling the IRQ, read the `smp_affinity` file
```
Comment (AnneCYH) [inline docs/how-to/tune-irq-affinity.rst:63]:
```suggestion
To check which CPUs are set for handling the IRQ, read the ``smp_affinity`` file
```

To align the formatting at line 72.

File: docs/how-to/tune-irq-affinity.rst [hunk 3]
```diff
@@ -60,22 +60,22 @@ In this case, the system has 20 CPUs available.
...
 
-OBS: The ``smp_affinity`` file is a bitmask, where each bit represents a CPU, where
+The ``smp_affinity`` file is a bitmask, where each bit represents a CPU, where
 ``1`` means that the CPU is allowed to handle the IRQ and ``0`` means that the CPU
 is not allowed to handle the IRQ. An output of ``ffffff`` means that all CPUs are
 allowed to handle the IRQ.
 
-Or you can use the :download:`check_irqs.sh` script to list the all the IRQs 
-associated  with a given CPU:
+To list all the IRQs associated with a given CPU, can use the
```
Comment (AnneCYH) [inline docs/how-to/tune-irq-affinity.rst:77]:
```suggestion
To list all the IRQs associated with a given CPU, you can use the
```

File: docs/how-to/tune-irq-affinity.rst [hunk 4]
```diff
@@ -107,20 +107,29 @@ associated  with a given CPU:
...
+
+        $ cat /proc/irq/0/smp_affinity_list
+
+        0-12,14-19
 
-    $ cat /proc/irq/0/smp_affinity_list
 
-    fdfff
+
+Do this process for all the IRQs that are being handled by the CPUs that
```
Comment (AnneCYH) [inline docs/how-to/tune-irq-affinity.rst:131]:
```suggestion
Do this for all the IRQs that are being handled by the CPUs that
```

**Review by AnneCYH — APPROVED** [review]
LGTM, thank you

**Pull request merged**